### ISSUE TITLE
Out of memory error found when running tests

### ISSUE THREAD

**Issue by superm1** [fwts/fwts#4]
When running FWTS on a pre-release system, the team reported the following critical error shows up; but there isn't an obvious cause.
```
method: Detected error 'Out of memory' when evaluating '\_PTS'.
```
There's a number of those types of errors.

It happens with the latest tool version from the PPA: `Version V23.03.00 (2023-03-27 04:12:10).`

The launch command was simply `sudo fwts`.

Attached full logs:
[fwts results](https://github.com/fwts/fwts/files/11314870/results.log)
[linux kernel log](https://github.com/fwts/fwts/files/11314872/kernel.log)
[acpidump](https://github.com/fwts/fwts/files/11314874/acpidump.log)

**Comment by Ivanhu5866** [comment]
Is this issue only on V23.03.00? could you check with V23.01.00?

I cannot reproduce on the machines that I have.
And test with the acpidump you attached, still cannot reproduce it.
$sudo fwts --dumpfile=acpidump.log

**Comment by superm1** [comment]
It reproduced with both versions, but an interesting finding is that it only happens when an addin card is put into system.  I guess that means that perhaps BIOS has a memory mappings error and this is how it's being exposed.  Agree?

**Comment by superm1** [comment]
It's found that this is fixed by increasing `MAX_SEMAPHORES` to a larger value, I sent up pull request for it.

**Comment by Ivanhu5866** [comment]
Hi superm1,
Thank you for your contribution. I appreciate your patch. Could you please assist in submitting the patch to the fwts-devel mailing list ([fwts-devel@lists.ubuntu.com](mailto:fwts-devel@lists.ubuntu.com))? All patches undergo review and ack on the mailing list.

**Comment by Ivanhu5866** [comment]
And I filed a bug on launchpad for tracking this issue.
https://bugs.launchpad.net/fwts/+bug/2025193

**Comment by superm1** [comment]
> Hi superm1, Thank you for your contribution. I appreciate your patch. Could you please assist in submitting the patch to the fwts-devel mailing list ([fwts-devel@lists.ubuntu.com](mailto:fwts-devel@lists.ubuntu.com))? All patches undergo review and ack on the mailing list.

I tried, but I got moderated.  Hope you can accept the PR instead or do manual fixup if it can't be released from moderation.

**Comment by Ivanhu5866** [comment]
fix released on fwts V23.07.00

**Comment by superm1** [comment]
> fix released on fwts V23.07.00

I don't see it pushed to GitHub

**Comment by Ivanhu5866** [comment]
pushed...
